src/test/java/unit/CloneWorkspace.java

```diff
@@ -0,0 +1,232 @@
+    // Add a referenced resource
+    externalDataset = ExternalBQDatasets.createDataset();
+  }
+
+  @AfterEach
+  public void cleanupEachTime() throws IOException {
+    workspaceCreator.login();
+    if (sourceWorkspace != null) {
+      TestCommand.Result result =
+          TestCommand.runCommand(
+              "workspace", "delete", "--quiet", "--workspace=" + sourceWorkspace.id);
+      sourceWorkspace = null;
+      if (0 != result.exitCode) {
+        logger.error("Failed to delete source workspace.");
+      }
+    }
+    if (destinationWorkspace != null) {
+      TestCommand.Result result =
+          TestCommand.runCommand(
+              "workspace", "delete", "--quiet", "--workspace=" + destinationWorkspace.id);
+      destinationWorkspace = null;
+      if (0 != result.exitCode) {
+        logger.error("Failed to delete destination workspace.");
+      }
+    }
+  }
+
+  @AfterAll
+  public static void cleanupOnce() throws IOException {
+    if (externalDataset != null) {
+      ExternalBQDatasets.deleteDataset(externalDataset);
+      externalDataset = null;
+    }
+  }
+
+  @Test
+  public void cloneWorkspace(TestInfo testInfo) throws Exception {
+    workspaceCreator.login();
+
+    // create a workspace
+    // `terra workspace create --format=json`
+    sourceWorkspace =
+        TestCommand.runAndParseCommandExpectSuccess(UFWorkspace.class, "workspace", "create");
+
+    // Add a bucket resource
+    UFGcsBucket sourceBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resource",
+            "create",
+            "gcs-bucket",
+            "--name=" + "bucket_1",
+            "--bucket-name=" + UUID.randomUUID(),
+            "--cloning=COPY_RESOURCE");
+
+    // Add another bucket resource with COPY_NOTHING
+    UFGcsBucket copyNothingBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resource",
+            "create",
+            "gcs-bucket",
+            "--name=" + "bucket_2",
+            "--bucket-name=" + UUID.randomUUID(),
+            "--cloning=COPY_NOTHING");
+
+    // Add a dataset resource
+    UFBqDataset sourceDataset =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFBqDataset.class,
+            "resource",
+            "create",
+            "bq-dataset",
+            "--name=dataset_1",
+            "--dataset-id=dataset_1",
+            "--description=The first dataset.",
+            "--cloning=COPY_RESOURCE");
+
+    ExternalBQDatasets.grantReadAccess(
+        externalDataset, workspaceCreator.email, ExternalBQDatasets.IamMemberType.USER);
+
+    // grant the user's proxy group access to the dataset so that it will pass WSM's access check
+    // when adding it as a referenced resource
+    ExternalBQDatasets.grantReadAccess(
+        externalDataset, Auth.getProxyGroupEmail(), ExternalBQDatasets.IamMemberType.GROUP);
```

Comment:
Can you do this in `setupOnce`, since the `workspaceCreator` is static? To get the proxy group email, you may need to `login()` first.